mem0/memory/graph_memory.py

```diff
@@ -106,6 +106,10 @@ def add(self, data, filters):
                 ADD_MEMORY_STRUCT_TOOL_GRAPH,
                 NOOP_STRUCT_TOOL,
             ]
+        elif self.llm_provider == "gemini":
+            # The `noop` ​​function n should be removed because it is unnecessary 
+            # and causes the error: "should be non-empty for OBJECT type"
+            _tools = [UPDATE_MEMORY_TOOL_GRAPH, ADD_MEMORY_TOOL_GRAPH]
```

Comment:
Trying to add a `noop` ​​function here seems unnecessary, and will return an error that the value in the object cannot be empty (from the Gemini related library).